### DISCUSSION TITLE
Roadmap Progress - Sept and October 2023 updates

### DISCUSSION THREAD

**Discussion by pchaurasia14** [dotnet/wpf#8488] [Announcements]
Below are the updates on 2023 Roadmap deliverables - 

## Modernization

| Area  | Update |
| ------------- |:-------------|
| Managed DWrite | We are glad to announce that we were able to review and **merge** another long standing [PR](https://github.com/dotnet/wpf/pull/6171) which enables **DWriteForwarder migration** 🎉🎉. Thank you @ThomasGoulet73 for your efforts and patience. 
| Win11 Theming | We have made some progress on this front. [Here](https://github.com/orgs/dotnet/projects/147) is the draft of the initial scope.<br/> We are working to refine this further. |
| New control - FolderBrowserDialog | Feature [merge](https://github.com/dotnet/wpf/pull/7244) is **completed**.<br/> [Sample application](https://github.com/microsoft/WPF-Samples/pull/500) and [tests](https://github.com/dotnet/wpf-test/pull/130) are available as well. <br /><br />Feature is now available in [.NET 8](https://dot.net). |
| Nullability annotations | Based on community [feedback](https://github.com/dotnet/wpf/discussions/7555#discussioncomment-5053915), the plan for Nullability annotations support is available [here](https://github.com/dotnet/wpf/issues/7563).<br/> Ready for community contributions. |
| Hardware acceleration in RDP | Feature [merge](https://github.com/dotnet/wpf/pull/7684) is **completed** for enabling hardware acceleration in RDP. This enables applications on RDP to take advantage of hardware acceleration capability (improved frames/second support, smoother animations, use of video memory, etc.) if the underlying system supports it. <br />Our testing on [sample applications](https://github.com/microsoft/WPF-Samples/tree/main/Animation/AnimationExamples) which uses animations showed an improvement of 10-30% sustained frame rate when running over RDP. <br/><br/>To enable the feature, you need to specify an appcontext switch(`System.Windows.Media.IsHardwareAccelerationEnabledForRdp`) in either .csproj or *.runtimeconfig.json . Do try this feature out and provide us with your feedback. <br /><br /> Feature is now available in [.NET 8](https://dot.net). |

---

## Infrastructure upgrades

| Area  | Update |
| ------------- |:-------------|
| Test Automation | No status change to report. We are working to add more unit tests to the project as well as fixing failing / flaky tests in the wpf-test repo. |
| Test Migration | Majority for feature tests and microsuites are now available in test repo (~ 80% completed) Few areas that are remaining have a higher amount on infrastructure dependency, hence we would migrate them as and when we incorporate changes to those feature areas. <br> - 2D [PR](https://github.com/dotnet/wpf-test/pull/203) <br/> - 3D [PR](https://github.com/dotnet/wpf-test/pull/204) <br/><br/> _Number of tests are indicative of total number of tests made available (including disabled tests)_ <br/> You can follow our progress overall in [this project](https://github.com/orgs/dotnet/projects/145).
---

## Issues/ PRs

| Area  | Update |
| ------------- |:-------------|
| PRs | Following PRs have been merged for September and October - <br/>  <br/> 1. Fix NullReferenceException in DynamicRenderer [#7552](https://github.com/dotnet/wpf/pull/7552) Thanks @lindexi ! <br/> 2. Cleanup TypeConverter replacements [#203](https://github.com/dotnet/wpf/pull/203) Thanks @hughbe ! <br/> 3. Added linker hint for DirectWriteForwarder [#4353](https://github.com/dotnet/wpf/pull/4353) Thanks @ThomasGoulet73 ! <br/> 4. Use built-in StringComparer.OrdinalIgnoreCase StringComparer [#8166](https://github.com/dotnet/wpf/pull/8166) Thanks @ThomasGoulet73 ! <br/> 5. Use ArgumentNullException.ThrowIfNull in PresentationUI and Themes [#8194](https://github.com/dotnet/wpf/pull/8194) Thanks @halgab ! <br/> 6. Use ArgumentNullException.ThrowIfNull in UIAutomation projects [#8148](https://github.com/dotnet/wpf/pull/8148) Thanks @halgab ! <br/> 7. Use ArgumentNullException.ThrowIfNull System.Windows.Controls.Ribbon and System.Windows.Input.Manipulations [#8193](https://github.com/dotnet/wpf/pull/8193) Thanks @halgab ! <br/> 8. Fix for Radio Button Behavior Across Disconnected Windows [#8202](https://github.com/dotnet/wpf/pull/8202)  <br/> 9. Closing the bitmapstream for UnknownBitMapDecoder [#8174](https://github.com/dotnet/wpf/pull/8174)  <br/> 10. Update translation state for ScrollBar_ContextMenu_Top resource [#8295](https://github.com/dotnet/wpf/pull/8295) <br/> 11. Fix event call in OpenFolderDialog.cs [#8299](https://github.com/dotnet/wpf/pull/8299) Thanks @adegeo ! <br/> 12. Put Unlock method within the finally block to ensure that the Lock and Unlock methods are called in pairs [#8157](https://github.com/dotnet/wpf/pull/8157) Thanks @lindexi ! <br/> 13. fix: EnsureDocumentStructure typo [#8321](https://github.com/dotnet/wpf/pull/8321) Thanks @0o001 !  <br/> <br/> We are very grateful to all the contributors who have helped improve WPF.<br/> We are aiming to reduce the total turnaround time for issues and PRs.  |

---

## Misc

| Area  | Update |
| ------------- |:-------------|
| Accessibility improvements | No major changes to report |
| ARM64 | Not yet started |

---

Going forward, our focus is primarily scoped to Win11 theming and BinaryFormatter obsoletion. Community contributions will be prioritized in these areas.

**Comment by MichelMichels** [comment]
Thanks to everyone for your contributions and hard work!
